src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,83 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String BUY = "buy";
+    private static final String SUPPLY = "supply";
+    private static final String COMMA = ",";
+    private static final String NEW_LINE = "\n";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] fileData = readFile(fromFileName).split(NEW_LINE);
+        String[] resultData = prepareStingToWrite(fileData);
+        clearFile(toFileName);
+        writeData(resultData, toFileName);
+    }
+
+    private String readFile(String fromFileName) {
+        String content = "";
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                content += line + NEW_LINE;
```

Comment:
you are using string concatenation (content += line + NEW_LINE) inside a loop. This can be inefficient for large files because strings in Java are immutable. Consider using a StringBuilder for better performance when concatenating strings in a loop.